lego/apps/users/models.py

```diff
@@ -377,6 +377,17 @@ def get_default_picture(self):
         else:
             return "default_other_avatar.png"
 
+    def delete(self, using=None, force=False):
+        from lego.apps.events import constants
+        from lego.apps.events.models import Event
+
+        if force:
+            for event in Event.objects.all():
+                if event.is_admitted(self):
```

Comment:
:open_mouth: :snail: 
```python
Event.objects.filter(registrations__user=self)
```
or something. The above filter at least gets only events where the user is registered. You could prob add a filter for admission as well :thinking:.

```python
Event.objects.filter(Q(registrations__user=self) & Q(registrations__pool__isnull=False))
```
Might work. (some quick testing seems good).

Reply:
See _Oh yes, speed._ commit. 🏎️